1.data_collection.py

```diff
@@ -0,0 +1,269 @@
+        status_code = response.status_code
+        reason_status = response.reason
+        results.append(response.json())
+        next_page = response.json()["meta"]["next_page"]
+        
+        params["page"] += 1
+
+        if status_code == 200:
+            print('The request was successful, the status code was returned:',status_code,' and as reason for the status code:',reason_status)
+        elif status_code == 402:
+            print('User could not be authorised, status code returned:', status_code,' and as reason for the status code: ',reason_status)
+        elif status_code == 404:
+            print('Something went wrong, the resource was not found, the status code was returned:', status_code,' and as reason for the status code: ',reason_status)
+        else:
+            print('Something unexpected has happened, the status code has been returned:', status_code,' and as reason for the status code: ',reason_status)
+
+    return f'There are {meta["current_page"]} pages. The status of the request is {status_code} and the reason of the state {reason_status}.'
+
+
+# ``Create one reusable python function called write_json_file that will write the responses to disk``
+
+# In[5]:
+
+
+# The above code is taking the results from the API call and writing them to a json file.
+def write_json_file(results, name_of_file): # You must give the result of the API call and the ubi and name of the file that will be created as a string
+
+    for i in results:
+
+        if len(results) <= 1:
+            print("There is only one page")
+
+        else:
+            call = json.dumps(i)
+            with open(name_of_file+".json", "a") as outfile:
+                outfile.write(call + "\n" )
+
+    return "Se ha creado el archivo:", name_of_file+".json"
+
+
+# ## Exercise 1.1: Get all teams. 
+# - https://www.balldontlie.io/home.html#get-all-teams
+# - Bear in mind pagination. It is possible to pass additional arguments to the API call.
+
+# In[6]:
+
+
+# url and params to the API call
+teams_url = 'https://www.balldontlie.io/api/v1/teams'
+teams_params = {} # It must be a dictionary
+
+
+# In[7]:
+
+
+# Function to the API call
+teams = paginate_api_calls(teams_url, teams_params)
+
+
+# In[8]:
+
+
+check_api_call(teams_url,teams_params)
+
+
+# In[9]:
+
+
+teams
+
+
+# In[10]:
+
+
+all_teams = write_json_file(teams,"notebooks/teams")
+
+
+# In[11]:
+
+
+all_teams
+
+
+# ## Exercise 1.2
+# - It is required to obtain all games of the 1991-1992 season. No need to rename/remove fields from the response.
+
+# In[12]:
+
+
+# url and params to the API call
+games_url = 'https://www.balldontlie.io/api/v1/games'
+games_params= {'seasons[]': 1991, "per_page":100} # It must be a dictionary
+
+
+# In[13]:
+
+
+# Function to the API call
+games = paginate_api_calls(games_url, games_params)
+
+
+# In[14]:
+
+
+check_api_call(games_url,games_params)
+
+
+# In[15]:
+
+
+games
+
+
+# In[16]:
+
+
+all_games_1991 = write_json_file(games,"notebooks/games")
```

Comment:
The exercise is asking you to write/read the JSON files in the `/data` folder, not in `/notebooks` folder